test/ResearcherContract.test.js

```diff
@@ -0,0 +1,130 @@
+const ResearcherContract = artifacts.require("ResearcherContract");
+const UserContract = artifacts.require("UserContract");
+
+const expectRevert = require("@openzeppelin/test-helpers").expectRevert;
+
+contract("ResearcherContract", (accounts) => {
+  let instance;
+  let userContract;
+  let [ownerAddress, resea1Address, resea2Address, resea3Address] = accounts;
+
+  const addResearcher = async (name, address) => {
+    await instance.addResearcher(
+      name,
+      "111.111.111-00",
+      "CPF",
+      "Brazil",
+      "SP",
+      "Jundiai",
+      "135465-005",
+      {from: address}
+    );
+  };
+
+  beforeEach(async () => {
+    userContract = await UserContract.new();
+
+    instance = await ResearcherContract.new(userContract.address);
+
+    await userContract.newAllowedCaller(instance.address);
+    await instance.newAllowedUser(resea1Address);
+
+  });
+
+  context("when will create new researcher (.addResearcher)", () => {
+    context("when is not an allowed user", () => {
+      it("should return error message", async () => {
+        await expectRevert(
+          addResearcher("Reseacher B", resea2Address),
+          "Not allowed user"
+        );
+      });
+    })
+
+    context("when is an allowed user", () => {
+      context("when researcher exists", () => {
+        it("should return error", async () => {
+          await addResearcher("Researcher A", resea1Address);
+          await expectRevert(
+            addResearcher("Researcher A", resea1Address),
+            "This researcher already exist"
+          );
+      });
```

Comment:
Esse fechamento não está alinhado com o `it` ao qual ele pertence

Reply:
Ok